code/modules/surgery/organs/blood.dm

```diff
@@ -70,13 +70,26 @@
 			if(BP.internal_bleeding)
 				internal_bleeding_rate += 0.5
 
-		bleed_rate = max(bleed_rate - 0.5, temp_bleed)//if no wounds, other bleed effects (heparin) naturally decreases
+		if(temp_bleed > 0 || bleed_rate > 0 )
+			if(temp_bleed)
+				bleed_rate = temp_bleed * get_bleed_modifier() //temp_bleed to bleed_rate
+			else
+				bleed_rate = max(0, bleed_rate - 0.5) //if no wounds, other bleed effects (heparin) naturally decreases and can not go below 0
```

Comment:
I'm very sorry, I spent way too much time being confused over this.

So, your previous version of `bleed_rate = max(bleed_rate - 0.5, temp_bleed * get_bleed_modifier()) ` was totally correct. Apologies for misleading your here.
The only "problem" with that one is if you swallow heparin, get to, say 4 bleed_rate, and get a tiny bit of calzeo after, then after calzeo wears off, heparin would start working its way back from zero. But that's arguably *intentional*, and not a problem whatsoever.

Current code however has an interesting bug in that heparin is completely ineffective if you have a minor bruise and are bleeding at a rate of 0.01 from it - the `temp_bleed` would get set to a small, yet nonzero value, and `bleed_rate` would be set to `temp_bleed` in turn, forgetting whatever changes heparin has done to it.

TL;DR: `bleed_rate = max(bleed_rate - 0.5, temp_bleed * get_bleed_modifier())` is the right way to handle things; I'm an idiot and I'm sorry.